README.md

```diff
@@ -26,30 +23,130 @@ dune-release help release # for help about releasing your package
+
+```
+dune-release tag v1.0.1
+```
+
+You can also directly use your VCS instead if the `dune-release tag` command does not fit your needs.
+
+If you need to delete a tag created by `dune-release`, use the following command:
+
+```
+dune-release tag -d v1.0.1
+```
+
+The full documentation of this command is available with
+```
+dune-release help tag
+```
+
+
+### Create the distribution archive
+
+Now that the release is tagged in your VCS, generate a distribution archive for it in the build directory with:
+
+```
+dune-release distrib
+```
+
+This uses the source tree of the HEAD commit for creating a distribution in the build directory. The distribution version string is the VCS tag description (e.g. `git-describe`) of the HEAD commit. Alternatively it can be specified on the command line.
+
+Basic checks are performed on the distribution archive when it is created, but save time by catching errors early. Hence test that your source repository lints and that it builds in the current build environment and that the package tests pass.
+
+```
+dune-release lint
+dune build # Check out the generated opam install file too
+dune runtest
+```
+
+The full documentation of this command is available with
+```
+dune-release help distrib
+```
+
 
-For instance:
+### Publish the distribution and documentation online
 
+Once the distribution archive is created you can now publish it and its documentation online.
+
+```
+dune-release publish
 ```
-$ dune-release help release
 
-NAME
-       dune-release - How to release a (dune) package
+You can publish the archive only with:
 
-DESCRIPTION
-       The basic release script is the following. Each step is refined and
-       explained with more details below.
+```
+dune-release publish distrib
+```
 
-       dune-release tag           # Tag the distribution with a version
-       dune-release distrib       # Create the distribution archive
-       dune-release publish       # Publish it on the WWW with its documentation
-       dune-release opam pkg      # Create an opam package
-       dune-release opam submit   # Submit it to OCaml's opam repository
+or publish the documentation only with:
+
+```
+dune-release publish doc
+```
 
-       The last four steps can be performed via a single invocation to
-       dune-release(1).
+The full documentation of this command is available with
+```
+dune-release help publish
 ```
 
+
+### Create an opam package
+
+The following steps still need the distribution archive created in the preceeding step to be in the build directory. If that's no longer the case but nothing moved in your VCS, you can simply invoke dune-release distrib, it should produce a bit-wise identical archive. If the VCS moved checkout the distribution commit to regenerate the archive or provide, in the subsequent commands, the archive manually via the `--dist-file` option.
```

Comment:
I think it might be confusing to explain all this distrib file details. how about explaining that the `pkg` subcommand generates an opam file to be used on opam-repo (ie that includes the DL URI for the distrib tarball and the tarball hash) and the `submit` subcommand creates the PR instead?

Reply:
done